hibernate-core/src/test/java/org/hibernate/test/onetoone/lazy/LazyToOneTest.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.onetoone.lazy;
+
+import org.hibernate.test.id.sequence.NegativeValueSequenceTest;
+import org.hibernate.testing.FailureExpected;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.MapsId;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import java.util.Date;
+
+import static org.hibernate.Hibernate.isInitialized;
+import static org.hibernate.testing.transaction.TransactionUtil.doInHibernate;
+import static org.junit.Assert.assertFalse;
+
+@TestForIssue(jiraKey = "HHH-12842")
+public class LazyToOneTest extends BaseCoreFunctionalTestCase {
+
+    private NegativeValueSequenceTest.PositiveTwoIncrementSize a;
+
+    @Override
+    protected Class<?>[] getAnnotatedClasses() {
+        return new Class<?>[] { Post.class, PostDetails.class };
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        doInHibernate( this::sessionFactory, s -> {
+            Post post = new Post();
+            post.setDetails(new PostDetails());
+            post.getDetails().setCreatedBy("ME");
+            post.getDetails().setCreatedOn(new Date());
+            post.setTitle("title");
+            s.persist(post);
+            s.flush();
+            s.clear();
```

Comment:
I don't think you need to flush() and clear(). Considering this init is specific to the test method, I would have moved it to the test method and would remove this setUp() altogether.

Reply:
Just wanted make sure that I would not get an actual object instead of the expected proxy because the entity was already in the session. But I think this can't happen anyway as both blocks use a different session.